500 error on image upload

I tried to upload Docker image but got 500 error with the stacktrace:
```
artipie         | [WARN] 2020-06-15 14:17:27,670 aws-java-sdk-NettyEventLoop-1-0 software.amazon.awssdk.http.nio.netty.internal.UnusedChannelExceptionHandler - A non-I/O exception occurred on a channel (5f86a3a7) that was not in use. The channel has been closed to prevent any ongoing issues.
artipie         | io.netty.channel.ChannelPipelineException: com.typesafe.netty.http.HttpStreamsHandler$1.handlerRemoved() has thrown an exception.
artipie         | 	at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:640)
artipie         | 	at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:477)
artipie         | 	at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:423)
   ...
artipie         | Caused by: java.lang.NoSuchMethodError: 'void com.artipie.asto.fs.RxFile.<init>(java.nio.file.Path, io.vertx.reactivex.core.file.FileSystem)'
artipie         | 	at com.artipie.docker.asto.AstoBlobs.lambda$put$5(AstoBlobs.java:115)
artipie         | 	at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback.onSuccess(SingleFlatMap.java:76)
artipie         | 	at io.reactivex.internal.observers.ResumeSingleObserver.onSuccess(ResumeSingleObserver.java:46)
artipie         | 	at io.reactivex.internal.operators.single.SingleJust.subscribeActual(SingleJust.java:30)
artipie         | 	at io.reactivex.Single.subscribe(Single.java:3603)
artipie         | 	at io.reactivex.internal.operators.single.SingleDelayWithCompletable$OtherObserver.onComplete(SingleDelayWithCompletable.java:69)
artipie         | 	at io.reactivex.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onComplete(CompletablePeek.java:115)
artipie         | 	at io.reactivex.internal.operators.flowable.FlowableFlatMapCompletableCompletable$FlatMapCompletableMainSubscriber.onComplete(FlowableFlatMapCompletableCompletable.java:165)
artipie         | 	at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler.runAndLogError(ResponseHandler.java:180)
artipie         | 	at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler.access$500(ResponseHandler.java:67)
artipie         | 	at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler$PublisherAdapter$1.onComplete(ResponseHandler.java:296)
artipie         | 	at com.typesafe.netty.HandlerPublisher.complete(HandlerPublisher.java:408)
artipie         | 	at com.typesafe.netty.HandlerPublisher.handlerRemoved(HandlerPublisher.java:395)
artipie         | 	at io.netty.channel.AbstractChannelHandlerContext.callHandlerRemoved(AbstractChannelHandlerContext.java:979)
artipie         | 	at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:637)
artipie         | 	... 49 more
```
It seems asto version is outdated, and it can be fixed by updating dependency to latest `0.22` version: https://github.com/artipie/asto/releases/tag/0.22

--- a/pom.xml
+++ b/pom.xml
@@ -62,7 +62,7 @@ SOFTWARE.
     <dependency>
       <groupId>com.artipie</groupId>
       <artifactId>asto</artifactId>
-      <version>0.18</version>
+      <version>0.22</version>
     </dependency>
     <dependency>
       <groupId>org.cactoos</groupId>

--- a/src/main/java/com/artipie/docker/asto/AstoBlobs.java
+++ b/src/main/java/com/artipie/docker/asto/AstoBlobs.java
@@ -33,8 +33,6 @@ import hu.akarnokd.rxjava2.interop.SingleInterop;
 import io.reactivex.Completable;
 import io.reactivex.Flowable;
 import io.reactivex.Single;
-import io.vertx.reactivex.core.Vertx;
-import io.vertx.reactivex.core.file.FileSystem;
 import java.io.IOException;
 import java.nio.channels.FileChannel;
 import java.nio.file.Files;
@@ -54,11 +52,6 @@ import java.util.concurrent.CompletionStage;
  */
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
 final class AstoBlobs implements BlobStore {
-
-    /**
-     * Vert.x file system used for temporary files.
-     */
-    private static final FileSystem FILE_SYSTEM = Vertx.vertx().fileSystem();
 
     /**
      * Storage.
@@ -112,7 +105,7 @@ final class AstoBlobs implements BlobStore {
             .andThen(Single.just(out))
             .flatMap(
                 ignored -> {
-                    final RxFile file = new RxFile(tmp, AstoBlobs.FILE_SYSTEM);
+                    final RxFile file = new RxFile(tmp);
                     return file.size().flatMap(
                         size -> SingleInterop.fromFuture(
                             this.asto.save(

--- a/src/test/java/com/artipie/docker/ExampleStorage.java
+++ b/src/test/java/com/artipie/docker/ExampleStorage.java
@@ -25,7 +25,6 @@ package com.artipie.docker;
 
 import com.artipie.asto.Storage;
 import com.artipie.asto.fs.FileStorage;
-import io.vertx.reactivex.core.Vertx;
 import java.net.URISyntaxException;
 import java.nio.file.Path;
 
@@ -40,7 +39,7 @@ public final class ExampleStorage extends Storage.Wrap {
      * Ctor.
      */
     public ExampleStorage() {
-        super(new FileStorage(path(), Vertx.vertx().fileSystem()));
+        super(new FileStorage(path()));
     }
 
     /**

--- a/src/test/java/com/artipie/docker/http/BlobEntityGetTest.java
+++ b/src/test/java/com/artipie/docker/http/BlobEntityGetTest.java
@@ -63,7 +63,7 @@ class BlobEntityGetTest {
     }
 
     @Test
-    void shouldReturnLayer() {
+    void shouldReturnLayer() throws Exception {
         final String digest = String.format(
             "%s:%s",
             "sha256",

--- a/src/test/java/com/artipie/docker/http/LargeImageITCase.java
+++ b/src/test/java/com/artipie/docker/http/LargeImageITCase.java
@@ -57,11 +57,7 @@ public final class LargeImageITCase extends AbstractDockerITCase {
     void setUp(@TempDir final Path temp) throws Exception {
         this.repo = this.startServer(
             temp,
-            new DockerSlice(
-                new AstoDocker(
-                    new FileStorage(temp, this.vertx().fileSystem())
-                )
-            )
+            new DockerSlice(new AstoDocker(new FileStorage(temp)))
         );
     }
 

--- a/src/test/java/com/artipie/docker/http/ManifestEntityGetTest.java
+++ b/src/test/java/com/artipie/docker/http/ManifestEntityGetTest.java
@@ -64,7 +64,7 @@ class ManifestEntityGetTest {
     }
 
     @Test
-    void shouldReturnManifestByTag() {
+    void shouldReturnManifestByTag() throws Exception {
         MatcherAssert.assertThat(
             this.slice.response(
                 new RequestLine("GET", "/base/v2/my-alpine/manifests/1", "HTTP/1.1").toString(),
@@ -84,7 +84,7 @@ class ManifestEntityGetTest {
     }
 
     @Test
-    void shouldReturnManifestByDigest() {
+    void shouldReturnManifestByDigest() throws Exception {
         final String hex = "cb8a924afdf0229ef7515d9e5b3024e23b3eb03ddbba287f4a19c6ac90b8d221";
         final String digest = String.format("%s:%s", "sha256", hex);
         MatcherAssert.assertThat(
@@ -140,7 +140,7 @@ class ManifestEntityGetTest {
     private static Matcher<Response> success(
         final String digest,
         final Key content
-    ) {
+    ) throws Exception {
         return new AllOf<>(
             Arrays.asList(
                 new RsHasStatus(RsStatus.OK),

